blockchain/My-Wallet-V3-iOS
# feat(Homebrew-Shapeshift) - IOS-1155 - New Exchange List Design

## Objective

This is the new HB exchange list screen.

## Description

This is a pretty large change but, the major things are:
* `ExchangeListViewController` now takes `ExchangeDependencies`. This struct will house all the services that all our exchange screens need. The intention is that the new exchange execution screens will also take this struct. 
* Added corresponding cells for the `ExchangeListViewController` to display trades
* Added a `Presenter`, `Interactor`, and protocols that stub out the APIs that restrict how the exchange list screen communicates with the various services and updates its interface. 

Note that there are some `// TODO` comments. This is somewhat of a WIP as it still needs to be wired up to the new exchange execution screen. Also, since we can't currently execute any exchanges on HB yet, we can't test refreshing or reloading this screen after trade execution. Lastly, there is still the overview screen to see the transaction. This hasn't been built yet. 

## How to Test

1. Add a bang here, in `ExchangeCoordinator`:
`if !isHomebrewAvailable` 
2. Make sure you've done some trades using Shapeshift. 
3. Build and run
4. Go to Exchanges
5. You should see the new exchange list screen. 

## Screenshot

![image](https://user-images.githubusercontent.com/41585563/44677541-d62da080-a9fb-11e8-9ce6-8e325593f098.png)

## Merge Checklist
